lib/wasi/src/state/mod.rs

```diff
@@ -957,18 +958,15 @@ impl WasiFs {
         Ok(cur_inode)
     }
 
-    /// Splits a path into the first preopened directory that is a parent of it,
-    /// if such a preopened directory exists, and the rest of the path.
-    ///
-    /// NOTE: this behavior seems to be not the same as what libpreopen is
-    /// doing in WASI.
-    ///
-    /// TODO: evaluate users of this function and explain why this behavior is
-    /// not the same as libpreopen or update its behavior to be the same.
+    /// Splits a path into the preopened directory that has the largest prefix of
+    /// the given path, if such a preopened directory exists, and the rest of the path.
```

Comment:
The wording of this comment confused me for a bit, this is just my perspective and additional data points should be gathered.

Either way it's a massive nitpick, feel free to ignore this.

With this though, my mind interperted the second sentence fragment "if such a preopened directory exists" to be the final part of the sentence, and then the "and the rest of the path" was confusing and didn't correctly link in my mind that it was talking about the second tuple value.

Instead, can I request that the topics of the "path of the preopened directory" and the "rest of the path" be located near eachother in the comment, with the error case being near the end of the line? That way it mirrors the structure of the result value, so reading the return value and comment is a similar read.